src/components/common/definitions/defineComponents.ts

```diff
@@ -1,9 +1,7 @@
-export const defineComponents = (...components: CustomElementConstructor[]) => {
-  for (const component of components) {
-    const tagName = (component as any).tagName;
+import type { IgniteComponent } from '../types';
 
-    if (tagName && !window.customElements.get(tagName)) {
-      window.customElements.define(tagName, component);
-    }
+export const defineComponents = (...components: IgniteComponent[]) => {
```

Comment:
So incoming types match `IgniteComponent` by schema alone?
I actually don't mind, spares us the need explicitly implement for each component

Reply:
Yep, it is a schema-based typing